src/programs/city/mine.js

```diff
@@ -100,7 +101,34 @@ class CityMine extends kernel.process {
       return
     }
 
-    const storage = this.room.storage
+    let storage = false
+    if (this.room.storage) {
+      storage = this.room.storage
+    } else if (this.room.terminal) {
+      storage = this.room.terminal
+    } else if (this.remote) {
+      const containers = this.room.structure[STRUCTURE_CONTAINER]
+      if (containers && containers.length > 0) {
+        if (containers.length > 1) {
+          containers.sort((a, b) => a.store[RESOURCE_ENERGY] - b.store[RESOURCE_ENERGY])
+        }
+        storage = containers[0]
+      } else {
+        const spawns = this.room.structure[STRUCTURE_SPAWN]
+        if (spawns) {
+          if (spawns.length > 1) {
+            spawns.sort((a, b) => a.energy - b.energy)
+          }
+          storage = containers[0]
```

Comment:
Is this line supposed to be `storage = spawns[0]`?

Reply:
that'd be dangerous, as spawns don't have a `.store` that the other processes would expect...

Reply:
I can't see anywhere that this variable is being checked for a `.store` property, but I guess it could be named something else since it won't always be a storage structure with this pull request.

Reply:
Yup, you're right.